torch/meta_registrations.py

```diff
@@ -0,0 +1,44 @@
+import torch
+
+Library = torch.library.Library
+aten = torch.ops.aten
+
+meta_lib = Library("aten", "IMPL", "Meta")
+
+# Implementations below are taken from https://github.com/albanD/subclass_zoo/blob/main/python_meta_tensor.py
+def meta_index_select(*args, **kwargs):
+    self, dim, index = args
+    assert not kwargs
+    result_size = list(self.size())
+    if self.dim() > 0:
+        result_size[dim] = index.numel()
+    return self.new_empty(result_size)
+
+def meta_inverse(*args, **kwargs):
+    (self,) = args
+    assert not kwargs
+    if self.numel() == 0:
+        return self.new_empty(self.size())
+    inverse = self.new_empty(self.size())
+    inverse.transpose_(-2, -1)
+    return inverse
+
+def meta_max(*args, **kwargs):
+    (self,) = args
+    assert not kwargs
+    return self.new_empty(())
+
+def meta_abs(*args, **kwargs):
+    (self,) = args
+    assert not kwargs
+    if self.is_complex():
+        from_complex = {torch.cfloat: torch.float, torch.cdouble: torch.double}
+        float_type = from_complex[self.dtype]
+        self.new_empty(self.size(), dtype=float_type)
+    else:
+        return self.new_empty(self.size())
+
+meta_lib.impl(aten.index_select.default, meta_index_select)
+meta_lib.impl(aten.inverse.default, meta_inverse)
+meta_lib.impl(aten.max.default, meta_max)
+meta_lib.impl(aten.abs.default, meta_abs)
```

Comment:
It would be nice if we had some decorator syntax for this. I propose something like:

```
@torch.library.impl(torch.ops.aten.index_select.default, "META")
def meta_index_select(...):
    ...
```

TBH people probably like the string syntax better because it's shorter: `"aten::index_select"`

Reply:
good idea! I added a `torch.library.impl` decorator with a similar API as that of `Library.impl` except that we also need to take the `Library` object as input here.

> TBH people probably like the string syntax better because it's shorter: "aten::index_select"

yeah this decorator is flexible so you can pass in either of `OpOverload` or `string` as the name argument